### PR TITLE
Fix aggregation of flaky test results in Jenkins

### LINKED ISSUE #8373: Flaky test detection is broken
**Describe the bug**

Some builds report flaky tests, but no failing test cases are reported in Jenkins. It's unclear whether or not the issue is with the plugin or with our Jenkins pipeline.

**To Reproduce**

Not sure. Here's an example build: https://ci.zeebe.camunda.cloud/blue/organizations/jenkins/camunda-cloud%2Fzeebe/detail/develop/1171/pipeline. I've saved the build forever, please delete it once this is fixed.

**Expected behavior**

Flaky tests are properly reported as separate failures for easier debugging.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR fixes the aggregation of flaky test results. There was an issue where I changed the glob pattern which caused flaky tests to be ignored when collecting the reports. Essentially, the flaky test plugin adds a `-FLAKY` suffix to the test name and creates a new report for that run under `surefire-reports` or `failsafe-reports`, so we were missing a glob `*` at the end of the pattern.

## Related issues

closes #8373 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
I've added 2 `DO NOT MERGE` commits which should be ignored, so please only review the first commit. These `DO NOT MERGE` commits are there to force flaky tests to occur and to run less tests. See the last build to see if/how it works :+1:

File: Jenkinsfile [hunk 1]
```diff
@@ -173,7 +173,7 @@ pipeline {
 
                     post {
                         always {
-                            junit testResults: "**/*/TEST*${SUREFIRE_REPORT_NAME_SUFFIX}.xml", keepLongStdio: true, allowEmptyResults: true
+                            junit testResults: "**/*/TEST*${SUREFIRE_REPORT_NAME_SUFFIX}*.xml", keepLongStdio: true, allowEmptyResults: true
```
Comment (korthout) [inline Jenkinsfile:176]: ❓ Would it work to do the junit task twice? Once for the normal test report and once for the flaky-test?

**Review by korthout — APPROVED** [review]
LGTM 🎄 

Just 1 question, but not blocking

**Comment by npepinpe** [comment]
Good point - it might be safer to run it multiple times, and also clearer why/what we're doing.

**Comment by npepinpe** [comment]
bors merge

**Comment by npepinpe** [comment]
bors retry

**Pull request merged**